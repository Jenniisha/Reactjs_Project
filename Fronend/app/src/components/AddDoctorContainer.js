import AddDoctor from "./AddDoctor";
import { connect } from 'react-redux';



// state is nothing but store that we are passing in the parameter
const mapStateToProps = (store) => {
    // console.log(store.reducer2);
    // console.log("Inside reducer2");

    return {
        doctorData: store.reducer2


    }
}

const mapDispatchToProps = (dispatch) => {

    return {

        addDoctor: (newDoctor) => dispatch({ type: "ADD_DOCTOR_TO_BACKEND", doctor: newDoctor }),
        editDoctor: (updatedDoctorObj) => dispatch({ type: "EDIT_DOCTOR_TO_BACKEND", doctor: updatedDoctorObj }),



    }

}



export default connect(mapStateToProps, mapDispatchToProps)(AddDoctor);