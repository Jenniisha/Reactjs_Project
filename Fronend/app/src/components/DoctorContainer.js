import SearchDoctor from "./SearchDoctor";
import { connect } from 'react-redux';



// state is nothing but store that we are passing in the parameter
const mapStateToProps = (store) => {
    // console.log(store.reducer1);
    // console.log("Inside reducer1");
    return {
        doctorData: store.reducer1


    }
}

const mapDispatchToProps = (dispatch) => {
    return {

        searchDoctor: (specialityValue) => dispatch({ type: "SEARCH_DOCTOR_FROM_BACKEND", speciality: specialityValue }),
        deleteDoctor: (doctNum) => dispatch({type:"DELETE_DOCTOR_FROM_BACKEND", doctorNumber: doctNum })
     



    }
}



export default connect(mapStateToProps, mapDispatchToProps)(SearchDoctor);