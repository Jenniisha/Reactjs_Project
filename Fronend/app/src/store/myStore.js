import { createStore, combineReducers, applyMiddleware } from 'redux';
import rootSaga from '../sagas/DoctorSaga';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';
import DoctorReducer from '../reducers/DoctorReducer';


const sagaMiddleware = createSagaMiddleware();


const rootReducer = combineReducers({
    reducer1: DoctorReducer,

});
const store = createStore(
    rootReducer, composeWithDevTools(applyMiddleware(sagaMiddleware))
    
);

export default store;

// middleware config is done! lets create a saga file and use it 
// run the saga middleware

sagaMiddleware.run(rootSaga);

//once the middleware runs, it will call our helloSaga.
//whenever any action is dispatched to any reducer

// //lets see the action here itself
// // i am going to dispatch any action. say "SEARCH_DOCTOR_SUCCESSFUL"action
// store.dispatch({type:"SEARCH_DOCTOR_SUCCESSFUL"});