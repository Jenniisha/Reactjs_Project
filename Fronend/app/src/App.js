import { Route, Routes } from "react-router";
import './App.css';
import NavBar from "./components/NavBar";

import DoctorContainer from "./components/DoctorContainer";
import store from "./store/myStore";
import { Provider } from 'react-redux';
import NoPageFound from "./components/NoPageFound";
import AddDoctorContainer from "./components/AddDoctorContainer";




function App() {
  return (
    <Provider store={store}>
      <Routes>

        <Route path="/" element={<Home />} />
        <Route path="/doctors/add" element={<AddDoctorContainer mode="ADD"/>} />
      <Route path="/doctors/edit/" element={<AddDoctorContainer mode="EDIT"/>} /> 
        <Route path="/doctors/search/speciality/" element={<DoctorContainer />} />
        <Route path="*" element={<NoPageFound />} />
      </Routes>


    </Provider>



  );
};

function Home() {
  return (
    <div>
      <NavBar /><br />
      <h1 style={{ textAlign: "center",color:"skyblue"}}>WELCOME TO  CLINIC MANAGEMENT SYSTEM</h1>
    </div>

  )
}


export default App;