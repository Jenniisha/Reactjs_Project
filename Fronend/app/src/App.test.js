import App from './App';
import {  mount, shallow } from 'enzyme';
import { render, screen } from '@testing-library/react';
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import tojson from 'enzyme-to-json';
import { useLocation } from "react-router";

import store from "./store/myStore";
import { Provider } from 'react-redux';
import AddDoctorContainer from './components/AddDoctorContainer';
import DoctorContainer from './components/DoctorContainer';

configure({ adapter: new Adapter() });


describe('test app renders or not', () => {
  test('Chck app comes without any issues', () => {
    shallow(<App />);
  });
});

//ADD DOCTOR TEST CASES
describe('test for add doctor', () => {
  test('App has a single AddDoctorContainer component and has somecontent ADD NEW DOCTOR FORM inside h1 tag', () => {
    const wrapper = mount(<Provider store={store}><AddDoctorContainer /></Provider>);
    const SomeContent = <h1 style={{ color: "skyblue", textAlign: "center" }}> ADD NEW DOCTOR FORM</h1>
    expect(wrapper.contains(SomeContent)).toEqual(true);
  });


  test('App renders the add doctor has 1 h1 element ', () => {
    var wrapper = mount( <Provider store={store}><AddDoctorContainer /></Provider>);
    //this is css selector
    const noOfH1Elements = wrapper.find('h1').length;//returns an array
    expect(noOfH1Elements).toEqual(1);
  });


  test('App renders the add doctor testing props are passed or not', () => {
    var wrapper = mount(<Provider store={store}><AddDoctorContainer name="Dr Joseph" /></Provider>);
    expect(wrapper.contains("Dr Joseph")).toEqual(false);
  });

  test("check addDoctorContainer comes without any issues", () => {
    mount(<Provider store={store}><AddDoctorContainer mode="ADD" /></Provider>);
  });

});


//SEARCH DOCTOR
describe('test for search doctor', () => {
  test('App has a single  search DoctorContainer component and has somecontent  DOCTOR SEARCH FORM inside h1 tag', () => {
    const wrapper = mount(<Provider store={store}><DoctorContainer /></Provider>);
    const SomeContent =  <h1>DOCTOR SEARCH FORM</h1>
    expect(wrapper.contains(SomeContent)).toEqual(true);
  });

  
  test('App renders the search doctor has 1 h1 element ', () => {
    var wrapper = mount( <Provider store={store}><DoctorContainer /> </Provider>);
    //this is css selector
    const noOfH1Elements = wrapper.find('h1').length;//returns an array
    expect(noOfH1Elements).toEqual(1);
  });


  test('App renders the search doctor testing props are passed or not', () => {
    var wrapper = mount(<Provider store={store}><DoctorContainer speciality="Orthodontist" /></Provider>);
    expect(wrapper.contains("Orthodontist")).toEqual(true);
  });

});

//EDIT DOCTOR
describe('test for edit doctor', () => {
  test('App has a single  edit DoctorContainer component ', () => {
    const wrapper = mount(<Provider store={store}><AddDoctorContainer mode="EDIT" qualification="MBBS"/></Provider>);
    expect(wrapper.contains(MBBS)).toEqual(true);
  });
});

//Grouping them into 1 test suit
//Snapshot
describe('Add Doctor snapshot', () => {
  test('Taking snapshot of App', () => {
    const tree = shallow(<Provider store={store}><App /></Provider>);
    expect(tojson(tree)).toMatchSnapshot();
  });


  test('Taking snapshot of SearchDoctor', () => {
    const tree = mount(<Provider store={store}><DoctorContainer /></Provider>);
    expect(tojson(tree)).toMatchSnapshot();
  });

});

