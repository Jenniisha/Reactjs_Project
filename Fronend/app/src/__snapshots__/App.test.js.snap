// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Doctor snapshot Taking snapshot of App 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <App />
</ContextProvider>
`;

exports[`Add Doctor snapshot Taking snapshot of SearchDoctor 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SearchDoctor)>
    <SearchDoctor
      deleteDoctor={[Function]}
      doctorData={Object {}}
      searchDoctor={[Function]}
    >
      <div>
        <h1>
          DOCTOR SEARCH FORM
        </h1>
        <form>
          <div
            style={
              Object {
                "border": "4px solid skyblue",
              }
            }
          >
            <label
              style={
                Object {
                  "margin": "20px",
                }
              }
            >
              SPECIALITY
            </label>
            <select
              name="speciality"
              onChange={[Function]}
            >
              <option
                value="Orthodontist"
              >
                Orthodontist
              </option>
              <option
                value="Opthalmologist"
              >
                Opthalmologist
              </option>
              <option
                value="Paediatrician"
              >
                Paediatrician
              </option>
              <option
                value="Gastroentrologist"
              >
                Gastroentrologist
              </option>
              <option
                value="Cardiologist"
              >
                Cardiologist
              </option>
            </select>
              
            <button
              onClick={[Function]}
            >
              SEARCH
            </button>
            <br />
            <br />
            <br />
            <table
              style={
                Object {
                  "border": "5px solid white",
                  "width": "100%",
                }
              }
            >
              <tr
                style={
                  Object {
                    "backgroundColor": "lightblue",
                    "color": "white",
                  }
                }
              >
                <th>
                  DOCTOR ID
                </th>
                <th>
                  DOCTOR NUMBER
                </th>
                <th>
                  NAME
                </th>
                <th>
                  QUALIFICATION
                </th>
                <th>
                  SPECIALITY
                </th>
                <th>
                  EDIT ACTION
                </th>
                <th>
                  DELETE ACTION
                </th>
              </tr>
            </table>
          </div>
        </form>
      </div>
    </SearchDoctor>
  </Connect(SearchDoctor)>
</Provider>
`;
